tests/poll.rs

```diff
@@ -201,6 +209,107 @@ fn registry_behind_arc() {
     handle2.join().unwrap();
 }
 
+/// Call all registration operations, ending with `source` being registered with `token` and `final_interests`.
+pub fn registry_ops_flow(
+    registry: &Registry,
+    source: &mut dyn Source,
+    token: Token,
+    init_interests: Interest,
+    final_interests: Interest,
+) -> io::Result<()> {
+    registry.register(source, token, init_interests).unwrap();
+    registry.deregister(source).unwrap();
+
+    registry.register(source, token, init_interests).unwrap();
+    registry.reregister(source, token, final_interests)
+}
+
+#[test]
+fn registry_operations_are_thread_safe() {
+    let (mut poll, mut events) = init_with_poll();
+
+    let registry = Arc::new(poll.registry().try_clone().unwrap());
+    let registry1 = Arc::clone(&registry);
+    let registry2 = Arc::clone(&registry);
+    let registry3 = Arc::clone(&registry);
+
+    let barrier = Arc::new(Barrier::new(4));
+    let barrier1 = Arc::clone(&barrier);
+    let barrier2 = Arc::clone(&barrier);
+    let barrier3 = Arc::clone(&barrier);
+
+    let mut listener = TcpListener::bind(any_local_address()).unwrap();
+    let addr = listener.local_addr().unwrap();
+
+    // Expect that multiple register/deregister/reregister work fine on multiple
+    // threads. Main thread will wait before the expect_events for all other 3
+    // threads to do their work. Otherwise expect_events timeout might be too short
+    // for all threads to complete, and call might fail.
+
+    let handle1 = thread::spawn(move || {
+        registry_ops_flow(
+            &registry1,
+            &mut listener,
+            ID1,
+            Interest::WRITABLE,
+            Interest::WRITABLE.add(Interest::READABLE),
```

Comment:
Can you remove the `WRITABLE` again, it shouldn't be use with `TcpListener`.

Reply:
done